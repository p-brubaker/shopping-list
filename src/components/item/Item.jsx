import React, { useState } from 'react';

export default function Item({ item, onChange, onDelete }) {
  const [editMode, setEditMode] = useState(false);
  const [name, setName] = useState(item.name);

  return editMode ? (
    <div>
      <input value={name} onChange={(e) => setName(e.target.value)} />
      <button
        onClick={() => {
          onChange({ id: item.id, name });
          setEditMode(false);
        }}
      >
        Save
      </button>
      <button onClick={() => onDelete(item.id)}>Delete</button>
    </div>
  ) : (
    <div>
      <p>{item.name}</p>
      <button onClick={() => setEditMode(true)}>Edit</button>
      <button onClick={() => onDelete(item.id)}>Delete</button>
    </div>
  );
}
