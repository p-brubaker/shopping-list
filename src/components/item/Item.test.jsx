import Item from '../item/Item';
import { render, screen } from '@testing-library/react';

it('should render the addItem component', async () => {
  const fakeItem = { name: 'barley', id: 0 };

  const { container } = render(<Item item={fakeItem} />);

  screen.getByText('Edit');
  screen.getByText('Delete');
  expect(container).toMatchSnapshot();
});
