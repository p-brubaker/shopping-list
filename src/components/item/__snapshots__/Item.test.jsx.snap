// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the addItem component 1`] = `
<div>
  <div>
    <p>
      barley
    </p>
    <button>
      Edit
    </button>
    <button>
      Delete
    </button>
  </div>
</div>
`;
